File: arrays/array_methods/array_methods.go
package main

import "fmt"

func main() {
	// Array declaration and initialization
	numbers := []int{1, 2, 3, 4, 5}
	fmt.Println("Array:", numbers)

	// Append an element
	numbers = append(numbers, 6)
	fmt.Println("Appended Array:", numbers)

	// Copy an array/slice
	copiedNumbers := make([]int, len(numbers))
	copy(copiedNumbers, numbers)
	fmt.Println("Copied Array:", copiedNumbers)

	// Slice an array/slice
	slicedNumbers := numbers[1:4] // [2, 3, 4]
	fmt.Println("Sliced Array:", slicedNumbers)

	// Modify the slice
	// Modifying a sliced array in Go, reflect on original array, because a slice is just a copy of array references;
	slicedNumbers[0] = 99
	fmt.Println("Modified Slice:", slicedNumbers)
	fmt.Println("Original Array:", numbers)

	// Flatten the nested slice (1-level only)
	nested := [][]int{{1, 2}, {3, 4}, {5, 6}}
	fmt.Println("Nested array:", nested)
	flat := flattenLevel(nested)
	fmt.Println("Flattened array:", flat)
}

func flattenLevel(input [][]int) []int {
	var flat []int
	for _, inner := range input {
		flat = append(flat, inner...)
	}
	return flat
}

File: arrays/array_methods/arrayMethods.ts
function main() {
  // Array declaration and initialization
  let numbers: number[] = [1, 2, 3, 4, 5];
  console.log("Array:", numbers);

  // Push an element
  numbers.push(6);
  console.log("Pushed Array:", numbers);

  // Copy an array/slice
  const copiedNumbers: number[] = numbers.slice();
  console.log("Copied Array:", copiedNumbers);

  // Slice an array/slice
  const slicedNumbers: number[] = numbers.slice(1, 4); // [2, 3, 4]
  console.log("Sliced Array:", slicedNumbers);

  // Modify the slice
  slicedNumbers[0] = 99;
  console.log("Modified Slice:", slicedNumbers);
  console.log("Original Array:", numbers);

  // Flat and array
  const nested = [1, 2, [3, 4], [5, [6]]];
  const flat = nested.flat(2);
  console.log("Flattened:", flat);
}

main();
